tests/subsys/logging/kernel_event_logger/src/main.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+#include <zephyr.h>
+#include <logging/kernel_event_logger.h>
+#include <misc/printk.h>
+#include <misc/util.h>
+
+K_THREAD_STACK_DEFINE(threadA_stack, 1024);
+K_THREAD_STACK_DEFINE(threadB_stack, 1024);
+K_THREAD_STACK_DEFINE(event_logger_stack, 4096);
+
+K_SEM_DEFINE(sem_sync, 0, 1);	/* starts off "not available" */
+
+#define UNKNOWN_EVENT  0xaa
+
+static struct k_thread threadA_data;
+static struct k_thread threadB_data;
+static u32_t timestamp;
+static u8_t complete;
+char event_type[][12] = {"READY_Q", "PEND_STATE", "EXIT_STATE"};
+
+static void event_logger(void)
+{
+	u32_t event_data[4];
+	u16_t event_id;
+	u8_t dropped;
+	u8_t event_data_size = (u8_t)ARRAY_SIZE(event_data);
+	int ret;
+
+	for (;;) {
+
+		ret = sys_k_event_logger_get_wait(&event_id,
+						  &dropped,
+						  event_data,
+						  &event_data_size);
+		if (ret < 0) {
+			continue;
+		}
+
+		timestamp = event_data[0];
+
+		switch (event_id) {
+		case KERNEL_EVENT_LOGGER_CONTEXT_SWITCH_EVENT_ID:
+			printk("tid of context switched thread = %x at "
+			       "time = %d\n", event_data[1], timestamp);
+			break;
+		case KERNEL_EVENT_LOGGER_INTERRUPT_EVENT_ID:
+			complete = 1;
+			break;
+		case KERNEL_EVENT_LOGGER_SLEEP_EVENT_ID:
+			complete = 1;
+			break;
+		case KERNEL_EVENT_LOGGER_THREAD_EVENT_ID:
+			printk("thread = %x, is moved to = %s ,at time = %d\n"
+			, event_data[1], event_type[event_data[2]], timestamp);
+			break;
+		default:
+			event_id = UNKNOWN_EVENT;
+			break;
+		}
+
+		zassert_not_equal(UNKNOWN_EVENT, event_id, "Unknown event");
```

Comment:
that is the only test I see in here, how are we verifying all events you have below?